pkg/autodiscovery/providers/env.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+const name = "dd_logs_config_custom_configs"
+
+// EnvProvider implements implements the ConfigProvider interface
+// It should be called once at the start of the agent.
+type EnvProvider struct{}
```

Comment:
nits: I am not a huge fan of `EnvProvider`, what about `EnvVarProvider` or `EnvironmentVariableProvider` ?